tools/serverpod_cli/bin/serverpod_cli.dart

```diff
@@ -203,205 +242,184 @@ Future<void> _main(List<String> args) async {
-        exit(1);
-      }
+    int priority;
+    var packageType = config.type;
+    switch (packageType) {
+      case PackageType.internal:
+        priority = 0;
+        break;
+      case PackageType.module:
+        priority = 1;
+        break;
+      case PackageType.server:
+        priority = 2;
+        break;
+    }
 
-      int priority;
-      var packageType = config.type;
-      switch (packageType) {
-        case PackageType.internal:
-          priority = 0;
-          break;
-        case PackageType.module:
-          priority = 1;
-          break;
-        case PackageType.server:
-          priority = 2;
-          break;
-      }
+    var generator = MigrationGenerator(
+      directory: Directory.current,
+      projectName: projectName,
+    );
 
-      var generator = MigrationGenerator(
-        directory: Directory.current,
-        projectName: projectName,
+    if (repair) {
+      await generator.repairMigration(
+        tag: tag,
+        force: force,
+        runMode: mode,
+        verbose: verbose,
       );
+    } else {
+      await generator.createMigration(
+        tag: tag,
+        verbose: verbose,
+        force: force,
+        priority: priority,
+      );
+      print('Done.');
+    }
 
-      if (repair) {
-        await generator.repairMigration(
-          tag: tag,
-          force: force,
-          runMode: mode,
-          verbose: verbose,
-        );
-      } else {
-        await generator.createMigration(
-          tag: tag,
-          verbose: verbose,
-          force: force,
-          priority: priority,
-        );
-        print('Done.');
-      }
+    return;
+  }
 
-      _analytics.cleanUp();
-      return;
-    }
+  if (results.command!.name == cmdLanguageServer) {
+    await runLanguageServer();
+    return;
+  }
 
-    if (results.command!.name == cmdLanguageServer) {
-      await runLanguageServer();
+  // Generate pubspecs command.
+  if (results.command!.name == cmdGeneratePubspecs) {
+    if (results.command!['version'] == 'X') {
+      print('--version is not specified');
       _analytics.cleanUp();
-      return;
+      throw ExitException(ExitCodeType.commandInvokedCannotExecute);
     }
+    performGeneratePubspecs(
+        results.command!['version'], results.command!['mode']);
+    return;
+  }
 
-    // Generate pubspecs command.
-    if (results.command!.name == cmdGeneratePubspecs) {
-      if (results.command!['version'] == 'X') {
-        print('--version is not specified');
-        _analytics.cleanUp();
-        return;
-      }
-      performGeneratePubspecs(
-          results.command!['version'], results.command!['mode']);
+  // Analyze pubspecs command.
+  if (results.command!.name == cmdAnalyzePubspecs) {
+    bool checkLatestVersion = results.command!['check-latest-version'];
+    if (!await pubspecDependenciesMatch(checkLatestVersion)) {
       _analytics.cleanUp();
-      return;
+      throw ExitException(ExitCodeType.general);
```

Comment:
It may be cleaner to leave out the `ExitCodeType.general` here (and in other places) as it is the default.

Reply:
Removed them all.